docs/2.5. APIs - Query Parameters.md

```diff
@@ -12,60 +12,65 @@ Query APIs SHOULD support pagination of their API resources where the 'paged' tr
+- Paging based on 'creation' time from the start of a collection provides the best mechanism to identify all resources held in a registry.
+- Paging based on 'update' time from the end of a collection provides the best mechanism to watch for changing resources in the case that WebSocket subscriptions are not available.
 
-* Paging through a collection for a second time, using the same cursors as previously should never result in new data appearing in the response payloads when using 'creation' ordering, provided the requested paging times are less than or equal to the maximum paging time held in the registry. However, note that as the registry is dynamic and resources may be updated or deleted a given cursor may return fewer results than it did previously. When using 'update' ordering and a query filter, new data may appear when cursors are used for a second time.
+Paging through a collection for a second time, using the same cursors as previously SHOULD NOT result in new data appearing in the response payloads when using 'creation' ordering, provided the requested paging times are less than or equal to the maximum paging time held in the registry. However, note that as the registry is dynamic and resources MAY be updated or deleted a given cursor MAY return fewer results than it did previously. When using 'update' ordering and a query filter, new data MAY appear when cursors are used for a second time.
 
-* When query parameters which perform filtering are used at the same time as paging, the filters must be applied by the implementation before applying paging parameters to the resulting data set.
+When query parameters which perform filtering are used at the same time as paging, the filters MUST be applied by the implementation before applying paging parameters to the resulting data set.
 
-* A server may choose its own default value for paging limit (see Example 1).
+A server MAY choose its own default value for paging limit (see Example 1).
 
-* Where both 'since' and 'until' parameters are specified, the 'since' value takes precedence where a resulting data set is constrained by the server's value of 'limit' (see Example 5).
+Where both 'since' and 'until' parameters are specified, the 'since' value takes precedence where a resulting data set is constrained by the server's value of 'limit' (see Example 5).
 
-* Servers should include 'prev' and 'next' Link headers and may include 'first' and 'last' Link headers in all responses.
+Servers SHOULD include 'prev' and 'next' Link headers and MAY include 'first' and 'last' Link headers in all responses.
```

Comment:
'first' and 'last' are optional but dead easy... first = `paging.since=0:0`, last = omit `paging.since` and `paging.until`

Reply:
Done